packages/gatsby-graphql-source-toolkit/README.md

```diff
@@ -0,0 +1,1057 @@
+  hasNextPage: boolean
+}
+
+interface IPaginationAdapter<TPage, TItem> {
+  name: string
+  expectedVariableNames: string[]
+  start(): IPageInfo
+  next(current: IPageInfo, page: TPage): IPageInfo
+  concat(acc: TPage, page: TPage): TPage
+  getItems(page: TPage): Array<TItem | null>
+}
+```
+
+Check out the `src/config/pagination-adapters` folder for examples.
+
+> Note: when setting `paginationAdapters` option you override built-in adapters completely
+> So if you want to be able to still use one of the existing adapters, pass them along with
+> your custom adapters:
+
+```js
+const { PaginationAdapters } = require("gatsby-graphql-source-toolkit")
+const MyCustomAdapter = {
+  // Your implementation
+}
+const config = {
+  // ... other options
+  paginationAdapters: PaginationAdapters.concat(MyCustomAdapter),
+}
+```
+
+## Tools Reference
+
+### Configuration Tools
+
+#### createDefaultQueryExecutor
+
+Creates default query executor suitable for [sourcing config](#query-executor):
+
+```ts
+function createDefaultQueryExecutor(
+  uri: string,
+  fetchOptions: FetchOptions,
+  queueOptions: PQueueOptions<any, any> = { concurrency: 10 }
+): IQueryExecutor
+```
+
+This implementation sends GraphQL requests using [`node-fetch`][14]
+(see library documentation for all possible `FetchOptions`). It also uses [p-queue][13]
+to restrict the number of concurrently executed queries.
+Default concurrency level is `10`, try increasing it to achieve higher throughput.
+
+#### wrapQueryExecutorWithQueue
+
+Takes existing query `executor` function and creates a new function with the same signature
+that runs with given concurrency level (`10` by default).
+
+```ts
+function wrapQueryExecutorWithQueue(
+  executor: IQueryExecutor,
+  queueOptions: PQueueOptions<any, any> = { concurrency: 10 }
+): IQueryExecutor
+```
+
+Under the hood it uses [p-queue][13] library to limit concurrency level, so refer to
+the library docs for all available `queueOptions`.
+
+#### loadSchema
+
+Executes [GraphQL introspection query][7] using provided [query executor](#query-executor)
+and creates an instance of [GraphQL Schema][15] using [`buildClientSchema`][16] utility from
+`graphql-js` package.
+
+```ts
+async function loadSchema(execute: IQueryExecutor): Promise<GraphQLSchema>
+```
+
+#### buildNodeDefinitions
+
+Simple utility that merges user-defined node type configs ([step 2](#2-configure-gatsby-node-types))
+with compiled queries ([step 4](#4-compile-sourcing-queries)) for every node type and produces
+a value suitable for `gatsbyNodeDefs` option of the [sourcing config](#configuration).
+
+```ts
+type RemoteTypeName = string
+
+interface IBuildNodeDefinitionArgs {
+  gatsbyNodeTypes: IGatsbyNodeConfig[]
+  documents: Map<RemoteTypeName, DocumentNode>
+}
+
+export function buildNodeDefinitions(
+  args: IBuildNodeDefinitionArgs
+): Map<RemoteTypeName, IGatsbyNodeDefinition>
+```
+
+### Query compilation tools
+
+#### generateDefaultFragments
```

Comment:
should not all function names in header (and in TOC) to be in code fences?


Example:
```suggestion
#### `generateDefaultFragments`
```